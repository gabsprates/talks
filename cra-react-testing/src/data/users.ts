export const users: UserType[] = [
  { name: "Albertinin Santos", github: "albertininm" },
  { name: "Bruno Carvalho", github: "bcarvalho89" },
  { name: "Gabriel Prates", github: "gabsprates" },
  { name: "Jared Gorski", github: "jaredgorski" },
  { name: "Mois√©s Menezes", github: "mgmenezes" },
  { name: "Ygor Costa", github: "ygorcosta" },
  { name: "Mateus Castro", github: "MateusCastro" },
];
