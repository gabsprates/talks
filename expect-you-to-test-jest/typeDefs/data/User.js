/**
 * @typedef {Object} UserData
 * @property {string} id
 * @property {string} name
 * @property {string} email
 */
