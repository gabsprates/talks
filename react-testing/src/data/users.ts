export const users: UserType[] = [
  { name: "Andr√© Simoes", github: "Andre-rsfc" },
  { name: "Artur Bani", github: "" },
  { name: "Carolina Sampaio", github: "carolinasamp" },
  { name: "Clarisse Silva", github: "" },
  { name: "Gabriel Ataide", github: "gabecar" },
  { name: "Gabriel de Paula", github: "gabriel949" },
  { name: "Gabriel Prates", github: "gabsprates" },
  { name: "Ian Rodrigues", github: "IanPaixao" },
  { name: "Izabela Bittencourt", github: "belabittencourt" },
  { name: "Pedro Barbosa", github: "" },
  { name: "Saulo Vinicius", github: "sauloviniciusf" }
];
