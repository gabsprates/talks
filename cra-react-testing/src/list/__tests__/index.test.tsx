import React from "react";
import { List } from "../index";
import { users } from "../../data/users";
import { render, fireEvent } from "@testing-library/react";

describe("Component: List", () => {
  it("should render an empty list", async () => {
    const wrapper = render(<List users={[]} onClick={jest.fn()} />);
    expect(wrapper.container.outerHTML).toMatchSnapshot();
    expect(wrapper.queryByText("nenhum dev encontrado"));
  });

  it("should render a fill list", () => {
    const wrapper = render(<List users={users} onClick={jest.fn()} />);

    expect(wrapper.container.outerHTML).toMatchSnapshot();
    expect(wrapper.container.querySelectorAll("li").length).toEqual(
      users.length
    );
  });

  it("should call spy function onClick li", () => {
    const spy = jest.fn();
    const wrapper = render(<List users={users} onClick={spy} />);

    expect(spy).not.toHaveBeenCalled();

    const itens = wrapper.getAllByRole("item-user");

    itens.forEach(userLi => {
      fireEvent.click(userLi);
    });
    expect(spy).toHaveBeenCalledTimes(itens.length);
  });
});
