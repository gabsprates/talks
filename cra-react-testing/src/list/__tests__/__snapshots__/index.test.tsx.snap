// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: List should render a fill list 1`] = `"<div><aside><ul><li><div class=\\"user\\"><strong class=\\"name\\">Albertinin Santos</strong><div class=\\"github\\">albertininm</div></div></li><li><div class=\\"user\\"><strong class=\\"name\\">Bruno Carvalho</strong><div class=\\"github\\">bcarvalho89</div></div></li><li><div class=\\"user\\"><strong class=\\"name\\">Gabriel Prates</strong><div class=\\"github\\">gabsprates</div></div></li><li><div class=\\"user\\"><strong class=\\"name\\">Jared Gorski</strong><div class=\\"github\\">jaredgorski</div></div></li><li><div class=\\"user\\"><strong class=\\"name\\">MoisÃ©s Menezes</strong><div class=\\"github\\">mgmenezes</div></div></li><li><div class=\\"user\\"><strong class=\\"name\\">Ygor Costa</strong><div class=\\"github\\">ygorcosta</div></div></li><li><div class=\\"user\\"><strong class=\\"name\\">Mateus Castro</strong><div class=\\"github\\">MateusCastro</div></div></li></ul></aside></div>"`;

exports[`Component: List should render an empty list 1`] = `"<div><div>nenhum dev encontrado ðŸ˜°</div></div>"`;
